service/frontend/workflowHandler.go

```diff
@@ -3022,6 +3180,51 @@ func (wh *WorkflowHandler) ListTaskListPartitions(ctx context.Context, request *
 	return resp, err
 }
 
+func (wh *WorkflowHandler) getRawHistory(
+	scope metrics.Scope,
+	domainID string,
+	execution gen.WorkflowExecution,
+	firstEventID int64,
+	nextEventID int64,
+	pageSize int32,
+	nextPageToken []byte,
+	transientDecision *gen.TransientDecisionInfo,
```

Comment:
This is not used. I think if transient decision is not nil, you will need to add fake decision scheduled and decision started event to the history. Please double check this part with @yux0